### ISSUE TITLE
s2n-quic test doesn't fail even after I change the code

### ISSUE THREAD

**Issue by danielsn** [model-checking/kani#1700]
```rust
fn read_table(x: u64) -> (u64, usize, u64) {
    debug_assert!(x <= MAX_VARINT_VALUE);

    macro_rules! table {
        ($(($two_bit:expr, $length:expr, $usable_bits:expr, $max_value:expr);)*) => {{
            let mut two_bit = 0;
            let leading_zeros = x.leading_zeros();
            $(
                two_bit += if leading_zeros < (64 - $usable_bits) {
                    1
                } else {
                    0
                };
            )*

            let len = 1 << two_bit;
            let usable_bits = len * 8 - 2;

            debug_assert_eq!(len as usize, encoding_size(x));

            (two_bit, len as usize, usable_bits)
        }};
    }

    table! {
        (0b00, 1, 6 , 63);
        (0b01, 2, 14, 16_373);
        (0b10, 3, 30, 1_073_741_823);
    }
}
```

```diff
     table! {
         (0b00, 1, 6 , 63);
-        (0b01, 2, 14, 16_383);
-        (0b10, 4, 30, 1_073_741_823);
+        (0b01, 2, 14, 16_373);
+        (0b10, 3, 30, 1_073_741_823);
     }
```

command was `cargo clean; time cargo kani --harness table_differential_test --enable-unstable --concrete-playback=print`

**Comment by danielsn** [comment]
Investigation shows that the macro doesn't actually use that value, so its not surprising that changing it doesn't cause an error
